modules/assim.batch/R/pda.emulator.R

```diff
@@ -194,13 +321,33 @@ pda.emulator <- function(settings, params.id=NULL, param.names=NULL, prior.id=NU
       
       ## Convert probabilities back to parameter values
       for(i in 1:n.param) {
-        m[,i] <- eval(prior.fn$qprior[prior.ind][[i]], list(p=mcmc.out[[c]][,i]))
+        m[,i] <- eval(prior.fn$qprior[prior.ind][[i]], list(p=mcmc.out[[c]]$mcmc[,i]))
       }
     }
-      colnames(m) <- pname[prior.ind]
-      mcmc.out[[c]] <- m
+    colnames(m) <- pname[prior.ind]
+    mcmc.list.tmp[[c]] <- m
+    
+    jvar.list[[c]] <- sqrt(mcmc.out[[c]]$jump@history[nrow(mcmc.out[[c]]$jump@history),])
   }
-
+  
+  
+  if(length(extension.check)==1 & !run.block){
+    
+    # merge with previous run's mcmc samples
+    mcmc.list <- mapply(rbind, mcmc.list, mcmc.list.tmp, SIMPLIFY=FALSE)
+    settings$assim.batch$iter <- nrow(mcmc.list[[1]])
+    burnin <- ifelse(!is.null(settings$assim.batch$burnin), 
+                     as.numeric(settings$assim.batch$burnin), 
+                     ceiling(min(2000,0.2*settings$assim.batch$iter)))
+    
+  }else{
+    
+    mcmc.list <- mcmc.list.tmp
+    burnin <- ifelse(!is.null(settings$assim.batch$burnin), 
+                     as.numeric(settings$assim.batch$burnin), 
+                     ceiling(min(2000,0.2*settings$assim.batch$iter)))
+  }
+  
   if(FALSE) {
     gp = kernlab.gp; x0 = init.x; nmcmc = settings$assim.batch$iter; rng= NULL; format = "lin"
```

Comment:
I missed this the first time around because it wasn't one of your commits, but in addition to @bpbond 's comment, I think the Google style guide (and most programming style guides that I've looked at) advise against placing multiple statements on a single line.